tools/lsp/preview.rs

```diff
@@ -1049,33 +1051,8 @@ pub fn reload_preview() {
     load_preview(pc, LoadBehavior::Load);
 }
 
-pub fn load_preview(preview_component: PreviewComponent, behavior: LoadBehavior) {
-    {
-        let mut cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
-        match behavior {
-            LoadBehavior::Reload => {
-                if !cache.ui_is_visible {
-                    return;
-                }
-            }
-            LoadBehavior::Load | LoadBehavior::BringWindowToFront => {
-                cache.set_current_component(preview_component)
-            }
-        }
-
-        match cache.loading_state {
-            PreviewFutureState::Pending => (),
-            PreviewFutureState::PreLoading => return,
-            PreviewFutureState::Loading => {
-                cache.loading_state = PreviewFutureState::NeedsReload;
-                return;
-            }
-            PreviewFutureState::NeedsReload => return,
-        }
-        cache.loading_state = PreviewFutureState::PreLoading;
-    };
-
-    let result = run_in_ui_thread(move || async move {
+fn reload_timer_function() -> impl std::future::Future<Output = ()> {
```

Comment:
Why not make this function async?

Reply:
Yeah... why not? I guess confusion :-) I tried a lot till I arrived here :-)

Reply:
Resolved in latest push.